### DISCUSSION TITLE
Missing value declared as output parameter

### DISCUSSION THREAD

**Discussion by prmunn** [nextflow-io/nextflow#4733] [Q&A] [answered]
## Bug report 

(Please follow this template replacing the text between parentheses with the requested information)

### Expected behavior and actual behavior

Val should be returned from a function

### Steps to reproduce the problem

Calling the following function:

process haplotagging_padUMI {

    output:
    val include_stagger, emit: include_stagger_val

    script:
        """
        include_stagger="Test value"

        echo \$include_stagger
        """
}

### Program output 

ERROR ~ Error executing process > 'haplotagging_create_bam_files:haplotagging_padUMI (1)'

Caused by:
  Missing value declared as output parameter: include_stagger

### Environment 

* Nextflow version:   latest version, using DSL2
* Java version:  Not sure
* Operating system:  Linux
* Bash version: (use the command `$SHELL --version`): GNU bash, version 5.1.8(1)-release (x86_64-redhat-linux-gnu)

### Additional context

(Add any other context about the problem here)

**Comment by robsyme** [comment, marked as answer]
Hi 

The `val` output type will emit a native Nextflow (Groovy) variable, and not a variable from the language in the script block (in your case, bash). If you would like to emit a bash variable, you would use the `env` type. See the documentation here: https://www.nextflow.io/docs/latest/process.html#output-type-env

**Comment by prmunn** [comment]
Ah... that appears to have fixed my problem. Thanks.
